### ISSUE TITLE
[RFC] Deprecate some alias functions in DatabaseQuery

### ISSUE THREAD

**Issue by HLeithner** [joomla-framework/database#174]
In the CMS we moved our method calls from `join('left')` to `leftJoin()` I'm not sure if this really makes sense because it's only a alias function to join() so we have small overhead.

Also it makes the API more complex and less maintainable.
Most of the functions are not in the interface which seams to make a problem in IntelliJ IDE so if 
we don't remove them they should be in the interface.

Functions for discussion:
[*]Join functions
[*]where functions

Functions missing in the interface
dateAdd
dateFormat
dump
escape (and alias)
quote (and alias)
quoteName (and alias)
format

Is there a reason why these functions are not in the interface?

**Comment by mbabker** [comment]
`escape()`, `quote()`, and `quoteName()` are intended to be done through the database driver primarily, I honestly don't know why they were added to the query class in the first place but they are all proxies to the driver so I didn't think they needed to be on the query interface (put them in one spot and leave it be).  Looks like `dateFormat()` is also a proxy.  The `dump()` method I'd just deprecate, it doesn't do anything useful on its own.  So `dateAdd()` and `format()` are probably just oversights.

The where and join shortcuts, personally I'm not a fan of littering interfaces with shortcut methods, which is why none of the interfaces have `e()`, `q()`, `qn()`, `*Join()`, `orWhere()`, and `andWhere()` methods.

**Comment by HLeithner** [comment]
Because we touch "every" query in the CMS we should do something soon. 
Maybe some input from @alikon and @wilsonge ?

btw. would it make sense to change the interface PreparableInterface return value of bind to $this the same why like in the other interfaces?

**Comment by mbabker** [comment]
> btw. would it make sense to change the interface PreparableInterface return value of bind to $this the same why like in the other interfaces?

Yeah.  Probably just missed it in one of the times I went over that stuff and replaced explicit names with `$this`.

**Comment by alikon** [comment]
not a big fan of changing something because of IDE (Intellij or whatever)
the alias like `e(), q(), qn()` can be deprecated as well
